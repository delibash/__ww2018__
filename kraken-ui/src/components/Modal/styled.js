import styled from 'styled-components';
import { colors } from './../../global/constants';

const StyledModal = styled.div`
  background: white;
  box-shadow: ${colors.BOX_SHADOW_ALL};
  position: fixed;
  top: 20%;
  left: 50%;
  transform: translate(-50%, -50%);
  padding: 2rem;
  z-index: 9999;
  nav {
    display: flex;
    justify-content: space-between;
    margin-top: 2rem;
  }
`;

export {
  StyledModal
};