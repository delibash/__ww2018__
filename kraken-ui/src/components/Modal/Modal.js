import React from 'react';
import PropTypes from 'prop-types';
import { StyledModal } from './styled';

const Modal = ({children, ...props}) => (
  <StyledModal
    {...props}
  >
    {children}
  </StyledModal>
);

Modal.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
    PropTypes.node
  ]).isRequired
};

export default Modal;
