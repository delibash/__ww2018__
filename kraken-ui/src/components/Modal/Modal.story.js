import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './../Button';
import Modal from './Modal';

storiesOf('Modal', module)
  .add('Default Modal', () => (
    <Modal
      style={{width: '20%'}}
    >
      <p>Modal Text</p>
      <nav>
        <Button onClick={action('cancel')}>Cancel</Button>
        <Button onClick={action('submit')}>Ok</Button>
      </nav>
    </Modal>
  ))

