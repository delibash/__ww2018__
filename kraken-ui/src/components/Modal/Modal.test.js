import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Modal from './Modal';

test('Default Modal', () => {
  const tree = renderer.create(<Modal>Modal</Modal>).toJSON();
  expect(tree).toMatchSnapshot();
});