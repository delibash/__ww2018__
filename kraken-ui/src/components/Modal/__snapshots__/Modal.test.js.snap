// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Modal 1`] = `
.c0 {
  background: white;
  box-shadow: 0 0 .5rem 0 rgba(0,0,0,0.5);
  position: fixed;
  top: 20%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  padding: 2rem;
  z-index: 9999;
}

.c0 nav {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 2rem;
}

<div
  className="c0"
>
  Modal
</div>
`;
